### PR TITLE
NetBSD: Fix build with LLVM-3.9

### PR DESCRIPTION
**Pull request by krytarowski**

### CODE REVIEW

**Comment by krytarowski** [comment]
@janvorli @jkotas PTAL

File: src/vm/amd64/jithelpers_fast.S [hunk 1]
```diff
@@ -110,7 +110,7 @@ LEAF_ENTRY JIT_WriteBarrier, _TEXT
         movabs  rax, 0xF0F0F0F0F0F0F0F0
 
         // Touch the card table entry, if not already dirty.
-        shr     rdi, 0Bh
+        shr     rdi, 0x0B
```
Comment (janvorli) [inline src/vm/amd64/jithelpers_fast.S]: Interesting, so now only one syntax for hex values is allowed in the clang assembler? I am just curious, what was the error message for the original syntax?

**Comment by janvorli** [comment]
LGTM

File: src/vm/amd64/jithelpers_fast.S (hunk 1 above)
Comment (krytarowski) [inline src/vm/amd64/jithelpers_fast.S]:
```
/tmp/pkgsrc-tmp/wip/coreclr-git/work/coreclr/src/vm/amd64/jithelpers_fastwritebarriers.S:35:21: error: unknown token in expression
        shr rdi, 0Bh
                    ^
/tmp/pkgsrc-tmp/wip/coreclr-git/work/coreclr/src/vm/amd64/jithelpers_fastwritebarriers.S:90:21: error: unknown token in expression
        shr rdi, 0Bh
                    ^
/tmp/pkgsrc-tmp/wip/coreclr-git/work/coreclr/src/vm/amd64/jithelpers_fastwritebarriers.S:128:21: error: unknown token in expression
        shr rdi, 0Bh
                    ^
/tmp/pkgsrc-tmp/wip/coreclr-git/work/coreclr/src/vm/amd64/jithelpers_fastwritebarriers.S:181:21: error: unknown token in expression
        shr rdi, 0Bh
                    ^
/tmp/pkgsrc-tmp/wip/coreclr-git/work/coreclr/src/vm/amd64/jithelpers_fastwritebarriers.S:252:21: error: unknown token in expression
        shr rdi, 0Bh
                    ^
/tmp/pkgsrc-tmp/wip/coreclr-git/work/coreclr/src/vm/amd64/jithelpers_fastwritebarriers.S:307:21: error: unknown token in expression
        shr rdi, 0Bh
                    ^
```

Comment (janvorli) [inline src/vm/amd64/jithelpers_fast.S]: Thanks a lot! I wonder if it is a bug or a feature :-). Anyways, it is good to stay consistent with hex format usage in the asm files, so this change is a good thing no matter what.

Comment (krytarowski) [inline src/vm/amd64/jithelpers_fast.S]: For some reason 0FFh is allowed and 0Bh rejected. I'm developing an LLDB plugin for NetBSD to track a process.

For me it looks like a regression in the compiler.. but I don't care about it as 0x is also a valid and used form.

**Comment by janvorli** [comment]
@mmitche - all OSX tests have failed with the same "no space left on device" error that you can see below.
http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/checked_osx_tst_prtest/1246/testReport/junit/CoreMangLib.system.delegate.miscellaneous/ClosedStatic/ClosedStatic/
Could you please take a look at the machine?

**Comment by mmitche** [comment]
@janvorli macpro-12 has plenty of space (146GB).  Either something running at the same time caused the issue (or run, or there is something wrong with coreclr.

**Comment by janvorli** [comment]
@mmitche It looks like a problem with the semaphores, e.g. exceeding the max number of those.
@mikem8361 - it looks like we have a problem with the debugger semaphores, likely exceeding the max allowed number of them. We are getting an error from sem_open:
sem_open(/clrct000047e60000000057289e56) failed: 28 (No space left on device)
I've googled a bit and found a discussion about the issue that seems to indicate that it can be due to max system-wide number of semaphores being exceeded. Is it possible that we are leaking semaphores?
https://forums.xilinx.com/t5/Embedded-Linux/sem-open-Failed-28-No-space-left-on-device/td-p/677198

**Comment by mikem8361** [comment]
Yes, it is possible to leak the named semaphores even though I go through great lengths to clean up them on termination.  If the coreclr process is killed via System.Diagnostic.Process.Kill/SIGKILL the semaphores and named pipe files are not cleaned up.

On linux the named semaphore files are in \dev\shm\sem.clr* (and the named pipes are in /tmp/clr*) if we want to clean them up in the CI lab as part of the job.

**Comment by janvorli** [comment]
The problem is that we are failing on OSX and semaphores on OSX are not exposed in the filesystem at all. So there seems to be no way to clean them except for rebooting the machine.

**Comment by krytarowski** [comment]
For your interest, there is a vulnerability in semaphore(2) usage on !NetBSD.

**Comment by krytarowski** [comment]
```
static const char* RuntimeStartupSemaphoreName = "/clrst%08x%016llx";
static const char* RuntimeOldContinueSemaphoreName = "/clrco%08x%016llx";
static const char* RuntimeContinueSemaphoreName = "/clrct%08x%016llx";
```

This is used to print integers into a string.. however it sets the minimum size with filled zeros, not the maximum. In case of larger numbers it will overflow the semaphore names. It may be only theoretical limit on these platforms, I addressed it on NetBSD with AND.

```
static uint64_t HashSemaphoreName(uint64_t a, uint64_t b)
{
        return (a ^ b) & 0xffffffff;
}
```

**Comment by janvorli** [comment]
@krytarowski I am not sure I understand - how can a 32 bit integer printed as hex overflow 8 characters?

**Comment by krytarowski** [comment]
Excuse me, I remembered wrongly, Unix platform defines DWORD as int.

**Comment by adityamandaleeka** [comment]
@dotnet-bot test OSX x64 Checked Build and Test

**Comment by adityamandaleeka** [comment]
Thanks @krytarowski

**Comment by krytarowski** [comment]
Thanks!

**Pull request merged**